app/forms/user_invite_form.rb

```diff
@@ -8,10 +8,12 @@ class UserInviteForm
   validate :validate_membership
   validates :service, :organisation, presence: true
 
-  def invite
-    return false unless valid?
-
-    UserInviteService.call(user, organisation) if save_user
+  def invite!
+    ActiveRecord::Base.transaction do
+      user.save!
+      membership.save!
+    end
+    UserInviteService.call(user, organisation)
```

Comment:
This should be in the transaction, otherwise the UserInviteService will trigger even if the the user hasn't been saved.

Reply:
It shouldn't.... If you put a debugger in it and call this in a way that raises an error, it won't hit the debugger. It will exit the method where the exception is raised.

Reply:
I believe `ActiveRecord::Base.transaction` captures exceptions, rolls back the transaction, and the re-raises the initial exception.